core/shared/src/main/scala/zio/internal/impls/FastLinkedQueue.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018-2024 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.internal
+
+import zio.Chunk
+import zio.stacktracer.TracingImplicits.disableAutoTrace
+
+import java.util.concurrent.ConcurrentLinkedQueue
+import scala.annotation.nowarn
+
+private[zio] final class FastLinkedQueue[A] extends MutableConcurrentQueue[A] with Serializable {
```

Comment:
Should we make this `private` instead of package-private instead?

Reply:
For consistency I feel like it should be the same

Reply:
Done